### PR TITLE
Add a cubed sphere

### PR DESCRIPTION
**Pull request by alexfikl**
As suggested, it just does the same thing as the simplex-based icosahedron generator.

### CODE REVIEW

**Comment by alexfikl** [comment]
And looks something like this

<p align="center">

</p>

File: meshmode/mesh/generation.py [hunk 1]
```diff
@@ -465,6 +465,29 @@ def generate_icosahedron(
             node_vertex_consistency_tolerance=node_vertex_consistency_tolerance,
             is_conforming=True)
 
+
+def generate_cube(r: float, order: int, *,
```
Comment (inducer) [inline meshmode/mesh/generation.py:469]: * `generate_cube_surface`?
* Add to docs?

File: meshmode/mesh/generation.py [hunk 2]
```diff
@@ -486,9 +510,20 @@ def generate_icosphere(r: float, order: int, *,
...
+    if issubclass(group_cls, SimplexElementGroup):
+        mesh = generate_icosahedron(r, order,
+                node_vertex_consistency_tolerance=node_vertex_consistency_tolerance,
+                unit_nodes=unit_nodes)
+    elif issubclass(group_cls, TensorProductElementGroup):
+        mesh = generate_cube(r, order,
+                node_vertex_consistency_tolerance=node_vertex_consistency_tolerance,
+                unit_nodes=unit_nodes)
+    else:
+        raise TypeError(f"unsupported 'group_cls': {group_cls}")
```
Comment (inducer) [inline meshmode/mesh/generation.py:526]: The name `icosphere` doesn't match this: The cubed sphere isn't derived from an icosahedron. :)

Rename as `generate_sphere`? Make deprecate `generate_icosphere` a deprecated wrapper?

**Review by inducer — COMMENTED** [review]
Thx! Few wrinkles below

File: meshmode/mesh/generation.py (hunk 1 above)
Comment (alexfikl) [reply meshmode/mesh/generation.py]: Done in 09dbdea

File: meshmode/mesh/generation.py (hunk 2 above)
Comment (alexfikl) [reply meshmode/mesh/generation.py:544]: Done in 88d184a.

**Review by inducer — COMMENTED** [review]
LGTM, thanks! I'll apply the one suggestion.

**Pull request merged**